clients/lsp-rust.el

```diff
@@ -1541,7 +1541,7 @@ and run a compilation"
               (`() (user-error "No compilation artifacts or obtaining the runnable artifacts failed"))
               (`(,spec) spec)
               (_ (user-error "Multiple compilation artifacts are not supported")))))
-         (list :type "cppdbg"
+         (list :type (if (featurep 'dap-gdb) "gdb" "cppdbg")
```

Comment:
Is `gdb` preferred over `cppdb`? 🤔

Reply:
I wasn't sure how else to do this.

As far as I understand it: If the user doesn't do `(require 'dap-gdb)` in their emacs config file then dap-gdb.el won't be used anyway (even though the file is on disk), right?

So this way should be backward compatible with whatever users have configured now.

On the other hand, if someone adds `(require 'dap-gdb)` to their emacs config now then lsb-rust will switch over to gdb.

I didn't think of more use cases than that yet.